Implement pop quiz system & fix Netlify issues

JoshHeng opened the pull request:
Added a quiz system to the Interactive book. Fixes #168 
See the README for more details or the [binary](https://deploy-preview-180--tender-ardinghelli-ebb79f.netlify.com/docs/binary.html) page for an example

~~This PR also fixes the Netlify issue with links not being relative~~

~~Note that the formatting is still strange for the quiz on the binary page. This is due to a problem with the interactive part above it, which I've already fixed in #174~~

# Screenshot
<img src="https://user-images.githubusercontent.com/34946235/71999908-221c6680-323a-11ea-9313-471923f715d3.png">

JoshHeng commented:
The new JS doesn't seem to be loading on Netlify. I'll provide screenshots

JoshHeng commented:
<img width="364" alt="2020-01-02_14-42 _3" src="https://user-images.githubusercontent.com/34946235/71999908-221c6680-323a-11ea-9313-471923f715d3.png">
See screenshot

JoshHeng commented:
I've also fixed the Netlify issue in this PR by changing all links to relative ones (instead of using the `learn.circuitverse.org` prefix).

This means that navigation now works and the correct css and js will be loaded

JoshHeng commented:
Note that the formatting is still strange for the quiz on the binary page. This is due to a problem with the interactive part above it, which I've already fixed in #174

JoshHeng commented:
(Preview at [https://deploy-preview-180--tender-ardinghelli-ebb79f.netlify.com/docs/binary.html](https://deploy-preview-180--tender-ardinghelli-ebb79f.netlify.com/docs/binary.html))

satu0king commented:
Hey, It looks good. However there seems to be a slight UI issue on my system. Can you look into it?

![image](https://user-images.githubusercontent.com/2092958/72268487-ea515c80-3647-11ea-9e3e-915769283a44.png)

JoshHeng commented:
> Hey, It looks good. However there seems to be a slight UI issue on my system. Can you look into it?
> 
> ![image](https://user-images.githubusercontent.com/2092958/72268487-ea515c80-3647-11ea-9e3e-915769283a44.png)

Hi, please refer to the first description/comment of the PR as this issue is addressed

aayush-05 commented on _layouts/default.html:
```diff
@@ -2,10 +2,8 @@
 
 <html lang="{{ site.lang | default: "en-US" }}">
 {% include head.html %}
-<script src="../assets/js/module.js"></script>
```
Any particular reason for removing jquery ?

JoshHeng commented:
Fixed, please review and merge

Note that currently the styling of the quiz won't work because of another issue which I've fixed in another PR. (#325)

To temporarily bypass this issue, open the inspector and replace `<link rel="stylesheet" href="https://learn.circuitverse.org/assets/css/just-the-docs.css">` with `<link rel="stylesheet" href="/assets/css/just-the-docs.css">`

mahmodHammad commented:
Cool idea 😀
Does this reusable every where?

Shivansh2407 commented:
> Cool idea 
> Does this reusable every where?

Yes, if this idea is reusable then we can add more quizzes in several pages to make this really interactive. Please confirm this @JoshHeng .

JoshHeng commented:
> 
> 
> Cool idea 😀
> Does this reusable every where?



> 
> 
> > Cool idea
> > Does this reusable every where?
> 
> Yes, if this idea is reusable then we can add more quizzes in several pages to make this really interactive. Please confirm this @JoshHeng .

Yes, see the README:
```
The quiz should be at the end of a page and is initialised with `{:.quiz}`. Questions are then added in a list, with correct answers being numbered subelements and incorrect answers being unordered subelements. Please see below for an example:
{:.quiz}
1. Example question
	1. Correct answer
	* Incorrect answer
2. What is 1 + 1?
	* 1
	1. 2	
   	* 3
Note that answers must be text only and not contain any other formatting.
```

JoshHeng commented:
@Shivansh2407 @mahmodHammad

Shivansh2407 commented:
Really nice @JoshHeng 👍
